### ISSUE TITLE
QTerminal window is without  borders (decorations)

### ISSUE THREAD

**Issue by Talkless** [trueos/lumina#460]
Using stable on VirtualBox 5.1.26 (with vboxvideo driver or whatever that was called):
![trueos_qterminal](https://user-images.githubusercontent.com/292191/29522323-4b967e3a-8678-11e7-9b3f-8fc5619ca700.png)

It is also a top most window for some reason, everything else is under it, inaccessible.

Other applications like xterm or QuipZilla works fine.

```
$ about

===================
General information
===================

boot environment now (N) … 12.0-CURRENT-up-20170820_140328 NR 2017-08-20
       after restart (R) … 12.0-CURRENT-up-20170820_140328 NR 2017-08-20
boot loader …………………………………… BSD
            type ……………………… BIOS
CPU ………………………………………………………… Intel(R) Core(TM) i7-4710HQ CPU @ 2.50GHz
    number of cores ……………… 8
host ……………………………………………………… TrueOS
memory ………………………………………………… 2999 MB available, 2127 MB free
OS git branch ……………………………………………………………………………………… trueos-stable
OS git revision ………………………………………………………………………………… ac2f0aa3b
OS kernel build time ………………………………………………………… Wed 2017 Jun 21 01:09:23 UTC
OS kernel identity …………………………………………… (uname -i) GENERIC
OS platform (architecture) ……………………… (uname -m) amd64
OS release level ………………………………………………… (uname -r) 12.0-CURRENT
OS version and patch level …… (freebsd-version) 12.0-CURRENT
TrueOS package set ………………… STABLE
TrueOS version …………………………… TrueOS-Desktop-201706210948
uptime ………………………………………………… 16 mins
user  …………………………………………………… vincas

More (TrueOS Desktop):

    desktop environment …… Lumina
    sound card driver ………… pcm0: <Intel ICH (82801AA)> (play/rec) default
    wireless driver ……………… 
    X11 drivers ………………………… vboxvideo_drv.so 
```

**Comment by Talkless** [comment]
I've seen this post:
https://forums.freebsd.org/threads/61181/#post-352450

> to the point you cannot use pkg(8) without the risk of breaking the system

And I actually installed something with it. Could this be an issue?

**Comment by beanpole135** [comment]
There is a bug in Fluxbox such that every time the resolution of the screen or number of screens is changed, then Fluxbox needs to be restarted to restore proper functionality. We already have a workaround for this in the latest version of Lumina from sources, or on the UNSTABLE package set for TrueOS.

**Comment by beanpole135** [comment]
There are no problems with using `pkg` on TrueOS - that is all the AppCafe does as well.

**Comment by Talkless** [comment]
True, if I disable "Auto-resize Guest Display" and keep it on 800x600, there are no issues.

> There are no problems with using pkg on TrueOS - that is all the AppCafe does as well.
Thanks for clarification.

**Comment by Unixware** [comment]
issue still exists *and* on current unstable version

**Comment by beanpole135** [comment]
Just to see if this still has the same cause: 
Run `killall fluxbox` and Lumina should automatically re-start Fluxbox for you. If that gives you back the window borders, then I will need details on *how* you resized the display, since it might be possible that something is directly interfering with the graphical settings (bypassing the safeties I put in) and triggering the same issue from Fluxbox.

**Comment by Unixware** [comment]
yes, pkill fluxbox brings back the window decoration (borders), I am running the TrueOS native - not under virtual env.

**Comment by beanpole135** [comment]
Which "version" of QTerminal did you run? I know a lot of people have been reporting that the "drop-down" option for QTerminal is broken in general (same window borders issue)

**Comment by Unixware** [comment]
thanks for the help, its the 0.8.0 and not the "drop down" style.
I am pretty sure I run the version with window decorations, they all start with borders then sometime the boarders were missing,  in that case I noticed that the QTerminal is always on front, can't be moved (eg using alt-tab) as background window while running other apps, even kill it and run a new still has no decorations -and stays on front, I need to pkill the fluxbox.

**Comment by beanpole135** [comment]
Yeah, so something is causing Fluxbox to go bonkers in the background then. Are you doing anything regarding screen resizes (whole session/screen, not just windows)? That is the main thing I have noticed that causes this.

**Comment by Unixware** [comment]
random incident I have to say, I did not notice it happened exactly after a specific task, I do use sometimes fullscreen apps (eg dosbox, or web video on fullscreen) anyway.

**Comment by beanpole135** [comment]
ok, just try to keep an eye on it and if it happens again make a note about what you did right beforehand. If you can find a way to reproduce it by narrowing down the steps, then we can try to track down the underlying config/change that is causing it.

**Comment by Unixware** [comment]
update:
the issue happened after run a fullscreen dosbox! any QTerm that will run after that , had no borders.

**Comment by beanpole135** [comment]
ok, dosbox might be manipulating the session/screen size when it goes into fullscreen mode then.
You might be able to find an option in dosbox to use the native resolution (or run it in a window) for the time being, and I will look into some extra safety checks for screen resizes.

**Comment by Unixware** [comment]
definitely fluxbox issue and is solved on (next version) 1.3.8 
https://sourceforge.net/p/fluxbox/bugs/1143/

**Comment by beanpole135** [comment]
thanks for verifying that!
I guess I will hold off on any significant surgery in Lumina to try and bypass this Fluxbox bug then.
